Iox #835 json config reader

grmplmpf opened the pull request:
## Pre-Review Checklist for the PR Author

1. [ ] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [ ] Tests follow the [best practice for testing][testing]
1. [ ] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [ ] Commits messages are according to this [guideline][commit-guidelines]
    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
    - [ ] Commit messages are signed (`git commit -s`)
    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [ ] Relevant issues are linked
1. [ ] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
<!-- Items in addition to the checklist below that the reviewer should look for -->

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] Unit tests have been written for new behavior
- [ ] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes **TBD**

elBoberido committed:
iox-#27 Generate puml svg recursively and multithreaded

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Update service-discovery diagrams

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Refine client and server options

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Refine response allocation

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Refine request and response header

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add simple plantuml diagram for request/response communication

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Update design document

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Update overview diagram

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add diagrams for typed and untyped API

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add diagrams for client and server port

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add diagrams to design document

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add RPC header to design document

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add RPC header to design document

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Update API with ctors

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add version info

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Fix typos

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Move request-response diagrams into own folder

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Download plantuml to a persistent location

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Refine typed API

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Further typed API refinement

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 implemented unique system id as basis for named mutex and semaphore

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#27 Refine untyped API

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Set default NUM_TREADS in plantuml script

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 implemented posix style ipc semaphores in windows

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 create ipc handle manager abstraction

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 interprocess capable mutex implemented

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 added windows handle to unix file handle translator

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 implemented filelock for windows

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 activating filelock tests

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 file lock file must be a valid file path

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 reduce the test so that it verifies that only at least the timeout is waited

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 shm unlink impossible/unnessary in windows

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 removed implicit shm removal from posix wrapper and packed it into explicit policy

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 removed implicit shm removal from posix wrapper and packed it into explicit policy

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 added shared memory tests

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 ipc channel works with windows and named pipes

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#27 Fix typo in typed API

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

dkroenke committed:
iox-#828 Set version number to future 2.0 release

elBoberido committed:
iox-#842 Use `@copydoc` instead of `@brief Implementation of`

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

dkroenke committed:
iox-#828 Add CMAKE_INSTALL_RPATH to iceoryx_meta

If iceoryx is built as shared lib we need to set an rpath
to the installed shared libs so that roudi can find them.

elBoberido committed:
iox-#842 Make dtor of MemoryProvider and MemoryBlock virtual

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#842 Use cxx::not_null for MemoryBlock::memoryAvailable parameter

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 adjusted platform settings variable namespace

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#842 Restrict access to some MemoryBlock methods

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 handle translator const correctness, adjusted ipc mutex name

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#842 Rework testing of protected method

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 Fix Windows handle_t Issue In HandleTranslator

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 introduce semaphores in named pipe

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#449 Add PoshRuntimeMock

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#449 Make PoshRuntimeMock usable for tests

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#449 Use PoshRuntimeMock for runtime factory tests

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#449 Ensure the RouDiEnvironment is not used with PoshRuntimeMock

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 Fix review findings

- Renamed CREATE_OR_OPEN into OPEN_OR_CREATE

- Add minimal doxygen docu to internal windows platform classes

- fix typos, reduced code doubling in unit tests

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Use new policy instead of ownership enum

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Implement named pipes with send/receive semaphore

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Adjust ownership handling

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#842 Update copyright and add noexcept

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#842 Rename memoryAvailable to onMemoryAvailable

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

grmplmpf committed:
iox-#835 Implemented a json config file reader for roudi

Signed-off-by: Schuenemann Oliver (XC-AD/ESB5) <oliver.schuenemann@de.bosch.com>

grmplmpf committed:
iox-#835 exclude JsonConfig Test from windows build

Signed-off-by: Schuenemann Oliver (XC-AD/ESB5) <oliver.schuenemann@de.bosch.com>

mossmaurice-apexai committed:
iox-#415 Add comments regarding refactoring

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add comments regarding refactoring

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add comments regarding refactoring

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elfenpiff committed:
iox-#743 Fix spelling mistakes in docu, adjust waitset examples

renamed waitset_sync into waitset_timer_driven_execution

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#743 fix syntax error

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#842 Fix documentation after rename

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
Merge pull request #845 from ApexAI/iox-#449-create-PoshRuntime-mock

Iox #449 create posh runtime mock

elfenpiff committed:
iox-#33 Fix review findings

- rename Policy into OpenMode

- return cxx::expected in unlinkIfExist in SharedMemory

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
Merge pull request #843 from ApexAI/iox-#842-cleanup-MemoryProvider-and-MemoryBlock

Iox #842 cleanup memory provider and memory block

elfenpiff committed:
iox-#33 Fix unused warning for gcc

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#743 Adapt asciinema recordings to new binary name and update admonition

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

grmplmpf commented:
Recreate Pull request to trigger actions

elfenpiff committed:
iox-#33 Fix shared memory moduletest

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Add platform setting for max shm name length

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Try to select right header for mac os SHM_NAME_MAX

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Use MAX_SHM_NAME_LENGTH value from windows in mac

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Increase SIGBUS_ERROR_MESSAGE_LENGTH

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Introduce iox_shm_unlink

- swap EINVAL errno with ENOENT in mac os

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Add missing mac os errno header

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#415 Remove two argument c'tor of ServiceDescription and adapt user
facing API

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove two argument c'tor of ServiceDescription and adapt user
facing API

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Adapt doxygen

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elfenpiff committed:
iox-#33 Address review findings

- corrected OpenMode translation

- added doxygen docu to Shared Memory

- cleaned up unlinkIfExists in SharedMemory

- oflags translation adjusted

- unlink() always resets the SharedMemory directly

- removing possible shm in SetUp of shared memory moduletests

- tested further side conditions for different OpenModes

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#415 Remove legacy f_*

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove legacy f_*

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elfenpiff committed:
iox-#33 Fix Timer timing test

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#415 Add isValid() checks where ServiceDescription is used

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Unify wildcard's by introducing capro::Wildcard

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Minor cleanups in capro::ServiceDescription tests

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove three integers in capro::ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove three integers in capro::ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#27 Add new queue policies

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove three integers in capro::ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#27 Add options client ports

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 Cleanup shared memory test

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
Merge pull request #849 from ApexAI/iox-#743-fix-docu-spelling-mistakes

iox-#743 Fix spelling mistakes in docu, adjust waitset examples

elBoberido committed:
iox-#27 Use client options in client port

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Get ChunkHeader pointer from user-header

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Move RpcHeader to own files and refine it

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
iox-#415 Adapt PoshRuntime mock and add new tests for invalid ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#27 Refine client port user

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

dkroenke committed:
iox-#828 Add unreleased section to Changelog

dkroenke committed:
iox-#828 Update PR Template and Contributing.md

Updated the template to add an entry in the Changelog if you close an
issue in GitHub. Additionally added that to Contributing.md

elBoberido committed:
iox-#27 Add serialization test for client options

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add test for RpcBaseHeader

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
iox-#33 Add copyright header, consistent variable naming

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#415 Verify validity of ServiceDescription and add related tests

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

dkroenke committed:
iox-#837 Add Release Script

This script search and replace the version numbers
in all necessary places of iceoryx.
As basis is the version number of the VERSION file used.

mossmaurice-apexai committed:
iox-#415 Allow wildcards only for searching ServiceDescription's

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Adapt copyright headers

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Update test case names

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Update copyright header

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Adapt PoshRuntimeMock to new API

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elfenpiff committed:
Merge pull request #834 from ApexAI/iox-#33-implement-ipc-mutex-semaphore

Iox #33 implement ipc mutex semaphore

elfenpiff committed:
iox-#33 Rebase with current master

Signed-off-by: Christian Eltzschig <me@elchris.org>

mossmaurice-apexai committed:
iox-#415 Use default c'tor to create invalid ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elfenpiff committed:
iox-#33 Add copyright header

Signed-off-by: Christian Eltzschig <me@elchris.org>

sschnitzer commented on iceoryx_posh/cmake/tiny_json/CMakeLists.txt.in:
```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
```
Is it correct that some files have Copyright 2021 and other Copyright 2020?

grmplmpf replied:
Fixed all copyright years to 2021

elfenpiff committed:
iox-#33 Simplify semaphore creation

Restore cout/cerr/clog flags when changing

Signed-off-by: Christian Eltzschig <me@elchris.org>

shankar-in commented:
@oschuene Guess you to want to make this PR to public Iceoryx? Currently this PR is targeting BoschGlobal Iceoryx. It has to point to the master of eclipse iceoryx?

elfenpiff committed:
iox-#33 removed unnecessary relative pointer from semaphore

Signed-off-by: Christian Eltzschig <me@elchris.org>

sschnitzer commented on iceoryx_posh/source/roudi/roudi_config_json_file_provider.cpp:
```diff
@@ -0,0 +1,181 @@
+    auto groupOfCurrentProcess = iox::posix::PosixGroup::getGroupOfCurrentProcess().getName();
+
+    json_t const* general;
+    general = json_getProperty(json, "general");
```
If I got the tiny_json API correctly, json_create() will return a nullptr if NUMBER_OF_JSON_NODES is too small. And json_getProperty does not check on nullptr, so potentially, this dereferences a nullpointer.

grmplmpf replied:
Added nullptr check

sschnitzer commented on iceoryx_posh/source/roudi/roudi_config_json_file_provider.cpp:
```diff
@@ -0,0 +1,181 @@
+        content.append(line);
+    }
+    json_t const* json = json_create(&content[0], mem, NUMBER_OF_JSON_NODES);
+    auto groupOfCurrentProcess = iox::posix::PosixGroup::getGroupOfCurrentProcess().getName();
```
Please place this line right before the for-loop it is used in. Improves readability.

sschnitzer commented on iceoryx_posh/include/iceoryx_posh/roudi/roudi_config_json_file_provider.hpp:
```diff
@@ -0,0 +1,77 @@
+class JsonRouDiConfigFileProvider : public iox::roudi::RouDiConfigFileProvider
+{
+  private:
+    constexpr static uint32_t NUMBER_OF_JSON_NODES = 1024;
```
This value seems to be somehow arbitrarily chosen. The TOML implementation does not have such a limit. If you really see the need to limit the stack memory usage, then please put this value into `iceoryx_posh_types.hpp ` and document it properly. If not (which is what I assume), just calculate the value using MAX_NUMBER_OF_MEMPOOLS and MAX_SHM_SEGMENTS, to make sure we neither prevent huge config files nor waste stack memory by using too high values.

sschnitzer commented on iceoryx_posh/test/moduletests/test_roudi_config_json_file_provider.cpp:
```diff
@@ -0,0 +1,402 @@
+                 "}\n";
+    ASSERT_EQ(iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_COUNT, parseJson(str).get_error());
+}
+
```
A test that tries to exceed NUMBER_OF_JSON_NODES is missing.

grmplmpf replied:
added

sschnitzer reviewed (changes requested):
See findings.

elfenpiff committed:
iox-#33 Use unnamed semaphores for named pipes

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Cleanup code, remove deprecated semaphore ctor, privatize getHandle in semaphore

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Wait until NamedPipe is initialized

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Removed unnecessary header

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Fix clang linker issue with static constexpr member

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Cleaned up shm creation

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 NamedPipeData is only destroyed when NamedPipe has the ownership

Signed-off-by: Christian Eltzschig <me@elchris.org>

elfenpiff committed:
iox-#33 Remove unused constants, clearify docu, use deadline timer

Signed-off-by: Christian Eltzschig <me@elchris.org>

elBoberido committed:
iox-#27 Add test for RequestHeader

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Add test for ResponseHeader

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elfenpiff committed:
Merge pull request #847 from ApexAI/iox-#33-efficient-named-pipes-with-semaphores

Iox #33 efficient named pipes with semaphores

dkroenke committed:
iox-#828 Add section for API changes to Changelog

Documenting API breaking changes in the Changelog
and give a hint in the PR template. Set the version number
to 1.90.0 for the upcoming pre-releases of iceoryx 2.0

mossmaurice-apexai committed:
Merge pull request #852 from ApexAI/iox-#415-refactor-service-description-rm-integers

Iox #415 Refactor Service Description Part1: Remove integers values

grmplmpf committed:
iox-#835 Calculate Number of Nodes from Segment and Mempool max

Signed-off-by: Schuenemann Oliver (XC-AD/ESB5) <oliver.schuenemann@de.bosch.com>

dkroenke committed:
Merge pull request #850 from ApexAI/iox-#828-backport-changelog-from-release_1.0

Iox #828 port changelog from release branch and add release helper script

grmplmpf committed:
Merge remote-tracking branch 'origin/master' into iox-#835-json-config-reader

mossmaurice-apexai committed:
Merge remote-tracking branch 'origin/master' into iox-#415-refactor-service-description-two-arg-ctor

mossmaurice-apexai committed:
iox-#415 Revert change in BasePortData and CaproMessage_test

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#27 Add const, noexcept and update doxygen

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Replace magic numbers with constexpr

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add first version for generic cxx::variant c'tor

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#27 Rework ClientOptions deserialization and add further tests

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

elBoberido committed:
iox-#27 Rename getFireAndForget to isFireAndForget

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
Merge pull request #853 from ApexAI/iox-#415-refactor-service-description-two-arg-ctor

Iox #415 Refactor Service Description Part2: Remove c'tor with two args

mossmaurice-apexai committed:
iox-#415 Fix cxx::variant tests by calling the right c'tor

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

dkroenke committed:
iox-#777 Set CMAKE_INSTALL_RPATH only when shared library is built

mossmaurice-apexai committed:
Merge remote-tracking branch 'origin/master' into iox-#415-refactor-service-description-restrict-wildcard-usage

dkroenke committed:
Merge pull request #862 from ApexAI/iox-#777-set-rpath-not-for-static-libs

iox-#777 Set CMAKE_INSTALL_RPATH only when shared library is built

mossmaurice-apexai committed:
iox-#415 Update doxygen

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Revert variant test cases by narrowing down overload resolution set of new c'tor

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
Merge pull request #819 from ApexAI/iox-#27-request-response-design-document

Iox #27 request response design document

elBoberido committed:
iox-#27 Update UML documentation

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove f_ in cxx::variant

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
Merge pull request #858 from ApexAI/iox-#27-client-port-implementation-and-test

Iox #27 client port implementation and test part 1

mossmaurice-apexai committed:
iox-#415 Add doxygen comments to cxx::variant

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Rename some cxx::variant test cases

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add std::decay to cxx::variant

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Replace some EXPECT's with ASSERT's

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add same type test case to cxx::variant tests

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Move true and false types to variant internal namespace

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Make all cxx::variant c'tor and assignment op's constexpr

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Rename InvalidString to InvalidIdString

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Wrap isValid() call in cxx::Expects

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Rename SearchMode to Any_t

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove using alias in ServiceRegistry

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

elBoberido committed:
iox-#865 Common puml file for common settings

Signed-off-by: Mathias Kraus <mathias.kraus@apex.ai>

mossmaurice-apexai committed:
iox-#415 Check isValid() in equality operator of ServiceDescription

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Document user-facing API changes in CHANGELOG.md

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
Merge pull request #864 from ApexAI/iox-#415-add-generic-ctor-to-variant

Iox #415 Add generic constructor to `cxx::variant`

mossmaurice-apexai committed:
Merge remote-tracking branch 'origin/master' into iox-#415-refactor-service-description-restrict-wildcard-usage

elBoberido committed:
Merge pull request #866 from ApexAI/iox-#865-common-puml-file-for-common-settings

iox-#865 Common puml file for common settings

mossmaurice-apexai committed:
iox-#415 Adapt findService() calls to new cxx::variant API

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove capturing of StdErr in PortManager test

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add wildcard example to changelog

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Check validity of ServiceDescription after serialisation and use cxx::Expects for Pub/Sub creation

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove superfluous wildcard definition

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Make changelog example clearer and adapt PoshRuntime test case as before

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Wrong user input shall not lead to termination of RouDi

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Extend changelog description of ServiceDescription change

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Add service.isValid() checks to PortManager::acquirePub/Sub methods

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
iox-#415 Remove wrong comment

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
Merge pull request #854 from ApexAI/iox-#415-refactor-service-description-restrict-wildcard-usage

Iox #415 Refactor Service Description Part3: Restrict wildcard usage to searching only

mossmaurice-apexai committed:
iox-#743 Add graphic showing polling alternatives

Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

mossmaurice-apexai committed:
Merge pull request #874 from eclipse-iceoryx/iox-#743-add-polling-alternatives-to-overview-md

iox-#743 Add graphic showing polling alternatives to docu

grmplmpf committed:
Merge remote-tracking branch 'origin/master' into iox-#835-json-config-reader

grmplmpf committed:
iox-#835 Calculate Max Nodes from settings

Signed-off-by: Schuenemann Oliver (XC-AD/ESB5) <oliver.schuenemann@de.bosch.com>
